### PR TITLE
Lau 666 retrieve stale users

### PR DESCRIPTION
**Pull request by rapolaskaseliscgi**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/LAU-663
https://tools.hmcts.net/jira/browse/LAU-666

### Change description ###

Added stale users retrieval and role assignments calls

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/test/java/uk/gov/hmcts/reform/idam/service/StaleUsersServiceTest.java [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+import static org.springframework.http.HttpStatus.INTERNAL_SERVER_ERROR;
+import static org.springframework.http.HttpStatus.OK;
+
+@ExtendWith(MockitoExtension.class)
+class StaleUsersServiceTest {
+    @Mock
+    RestClient client;
+
+    @Mock
+    private StaleUsersRequestParams requestParams;
```
Comment (MartinYSpasov) [inline src/test/java/uk/gov/hmcts/reform/idam/service/StaleUsersServiceTest.java:38]: Do we use this?

File: src/integrationTest/java/uk/gov/hmcts/reform/idam/ApplicationIntegrationTest.java [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+    @Test
+    void testShouldBootstrapSpringContext() {
+        WireMock.listAllStubMappings();
+        assertThat(Boolean.valueOf(FLAG)).isTrue();
+    }
+
+    @Test
+    void testShouldReturnUsersWithoutRoles() {
+        stubFor(get(urlPathEqualTo("/api/v2/staleUsers")).willReturn(
```
[the remark this answers is not in the document]
Comment (rapolaskaseliscgi) [reply src/integrationTest/java/uk/gov/hmcts/reform/idam/ApplicationIntegrationTest.java:44]: This is going to change once I've switched to cucumber tests, so I don't think it is worth addressing this at this time.

File: src/test/java/uk/gov/hmcts/reform/idam/service/StaleUsersServiceTest.java (hunk 1 above)
Comment (rapolaskaseliscgi) [reply src/test/java/uk/gov/hmcts/reform/idam/service/StaleUsersServiceTest.java:38]: Good spot. No it is not used, though I was thinking about and I guess it stayed.

File: src/main/java/uk/gov/hmcts/reform/idam/service/StaleUsersService.java [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+        );
+
+        if (response.getStatus() == OK.value()) {
+            var staleUsersResponse = response.readEntity(StaleUsersResponse.class);
+            if (!staleUsersResponse.getStaleUsers().isEmpty()) {
+                collected.addAll(staleUsersResponse.getStaleUsers());
+                staleUsersResponse
+                    .getStaleUsers()
+                    .stream()
+                    .forEach(user -> log.debug("Got stale user {}", user));
```
[the remark this answers is not in the document]
Comment (rapolaskaseliscgi) [reply src/main/java/uk/gov/hmcts/reform/idam/service/StaleUsersService.java:51]: This is at debug level, I can bump down to trace or remove entirely.

File: src/main/java/uk/gov/hmcts/reform/idam/service/UserRoleService.java [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+                .getRoleAssignments()
+                .stream()
+                .filter(assignment -> "IDAM".equals(assignment.getActorIdType()))
+                .map(RoleAssignment::getActorId)
+                .toList();
+            return staleUsers.stream()
+                .filter(userId -> !assignments.contains(userId))
+                .toList();
+        }
+        return List.of();
```
[the remark this answers is not in the document]
Comment (rapolaskaseliscgi) [reply src/main/java/uk/gov/hmcts/reform/idam/service/UserRoleService.java]: Addressed

**Review by MartinYSpasov — APPROVED** [review]

**Pull request merged**